action_server/src/robocorp/action_server/_server_expose.py

```diff
@@ -0,0 +1,113 @@
+import asyncio
+import logging
+import json
+import httpx
+
+import websockets
+from fastapi import FastAPI
+from fastapi.testclient import TestClient
+from pydantic import BaseModel
+from ._settings import get_settings
+
+log = logging.getLogger(__name__)
+
+
+class SessionPayload(BaseModel):
+    sessionId: str
+    sessionSecret: str
+
+
+class BodyPayload(BaseModel):
+    requestId: str
+    path: str
+    method: str = "GET"
+    body: dict | None = None
+
+
+def forward_request(client: TestClient, payload: BodyPayload) -> httpx.Response:
+    if payload.method == "GET":
+        return client.get(payload.path)
+    elif payload.method == "POST":
+        return client.post(payload.path, json=payload.body)
+    elif payload.method == "PUT":
+        return client.put(payload.path, json=payload.body)
+    elif payload.method == "DELETE":
+        return client.delete(payload.path)
+    else:
+        raise NotImplementedError(f"Method {payload.method} not implemented")
+
+
+async def expose_server(app: FastAPI):
+    """
+    Exposes the server to the world.
+    """
+
+    settings = get_settings()
+
+    async def listen_for_requests():
+        max_retries = 3
+        retry_delay = 1
+        retries = 0
+
+        session_payload: SessionPayload | None = None
+        while retries < max_retries:
+            try:
+                headers = (
+                    {
+                        "x-session-id": session_payload.sessionId,
+                        "x-session-secret": session_payload.sessionSecret,
+                    }
+                    if session_payload
+                    else {}
+                )
+
+                async with websockets.connect(
+                    f"wss://client.{settings.expose_url}", extra_headers=headers
+                ) as ws:
+                    while True:
+                        message = await ws.recv()
+
+                        data = json.loads(message)
+
+                        try:
+                            session_payload = SessionPayload(**data)
+                            log.info(
+                                f"🌍 URL: https://{session_payload.sessionId}.{settings.expose_url}/openapi.json"
```

Comment:
Updated the proxy server to resolve main path to `/openapi.json`:
```suggestion
                                f"🌍 URL: https://{session_payload.sessionId}.{settings.expose_url}"
```